### PR TITLE
xnviewmp: add version 0.93.1

### PR DESCRIPTION
**Pull request by Befod**

### CODE REVIEW

File: bucket/xnviewmp.json [hunk 1]
```diff
@@ -0,0 +1,44 @@
+{
+    "homepage": "https://www.xnview.com/en/xnviewmp/",
+    "description": "Photo viewer, image resizer & batch converter",
```
Comment (Ash258) [inline bucket/xnviewmp.json]:
```suggestion
    "description": "Photo viewer, image resizer & batch converter.",
```

File: bucket/xnviewmp.json [hunk 2]
```diff
@@ -0,0 +1,44 @@
+{
+    "homepage": "https://www.xnview.com/en/xnviewmp/",
+    "description": "Photo viewer, image resizer & batch converter",
+    "version": "0.93.1",
+    "license": "Freeware",
+    "architecture": {
+        "64bit": {
+            "hash": "d0c08ba12f253207064e74ebb5cd866410998031ace74a43dbbe01275ddeb179",
+            "url": "http://download.xnview.com/XnViewMP-win-x64.zip"
```
Comment (Ash258) [inline bucket/xnviewmp.json]: Put URL before hash.

File: bucket/xnviewmp.json [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    "bin": "XnViewMP.exe",
+    "shortcuts": [
+        [
+            "XnViewMP.exe",
+            "XnViewMP"
+        ]
+    ],
+    "persist": "xnview.ini",
+    "pre_install": [
+        "function CreateFile ($name) {",
```
Comment (Ash258) [inline bucket/xnviewmp.json]: Function is overkill for just one file
```suggestion
```

File: bucket/xnviewmp.json [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+        "    if (!(Test-Path \"$persist_dir\\$name\")) {",
```
Comment (Ash258) [inline bucket/xnviewmp.json]:
```suggestion
        "if (!(Test-Path \"$persist_dir\\xnview.ini\")) { New-Item \"$dir\\xnview.ini\" -ItemType File | Out-Null }"
```

File: bucket/xnviewmp.json [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+        "        New-Item -Force -Path \"$persist_dir\" -Name $name -ItemType \"file\"",
```
Comment (Ash258) [inline bucket/xnviewmp.json]:
```suggestion
```

File: bucket/xnviewmp.json [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+        "    }",
+        "}",
+        "CreateFile 'xnview.ini'"
```
Comment (Ash258) [inline bucket/xnviewmp.json]:
```suggestion
```

File: bucket/xnviewmp.json [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+            "XnViewMP"
+        ]
+    ],
+    "persist": "xnview.ini",
+    "pre_install": [
+        "function CreateFile ($name) {",
+        "    if (!(Test-Path \"$persist_dir\\$name\")) {",
+        "        New-Item -Force -Path \"$persist_dir\" -Name $name -ItemType \"file\"",
+        "    }",
+        "}",
```
Comment (Ash258) [inline bucket/xnviewmp.json]:
```suggestion
```

File: bucket/xnviewmp.json [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+            "XnViewMP.exe",
+            "XnViewMP"
+        ]
+    ],
+    "persist": "xnview.ini",
+    "pre_install": [
+        "function CreateFile ($name) {",
+        "    if (!(Test-Path \"$persist_dir\\$name\")) {",
+        "        New-Item -Force -Path \"$persist_dir\" -Name $name -ItemType \"file\"",
+        "    }",
```
Comment (Ash258) [inline bucket/xnviewmp.json]:
```suggestion
```

File: bucket/xnviewmp.json [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+        "}",
+        "CreateFile 'xnview.ini'"
+    ],
+    "extract_dir": "XnViewMP",
```
Comment (Ash258) [inline bucket/xnviewmp.json]: Add this after architecture property

File: bucket/xnviewmp.json (hunk 2 above)
Comment (linsui) [reply bucket/xnviewmp.json]: And use https in all url

**Comment by Befod** [comment]
Thanks a lot. 👍

**Comment by linsui** [comment]
`persist` doesn't work. You need to copy them manully. There are some .db files need persist. There is a problem, I do not know what "other flies" is and I didn't see them in my test. Please test what it is.

```
    "pre_install": [
        "if (!(Test-Path \"$persist_dir\" -PathType Container)) {",
        "    New-Item \"$persist_dir\" -ItemType Directory -Force | Out-Null",
        "    Set-Content \"$dir\\xnview.ini\" @('[Start]', 'otherFlag=1', 'dbFlag=1', 'dbThumbFlag=4') -Encoding ASCII -Force",
        "}",
        "else { Copy-Item \"$persist_dir\\*\" \"$dir\" -Include '*.ini', '*.db' -Force }"
    ],
    "uninstaller": {
        "script": "Copy-Item \"$dir\\*\" \"$persist_dir\" -Include '*.ini', '*.db' -Force"
    },
```

**Comment by Befod** [comment]
Files of "Folder for other files" are the configuration of  "toolbar" and "shortcuts".
`Tools` > `Settings` > `Interface` > `Toolbar` > `Load/Save`
`Tools` > `Settings` > `Interface` > `Shortcuts` > `Load/Save`
- default.bar
- default.keys

"xnview.ini" can only be stored in `User` and `Program`,but other files can use a custom directory.It is unnecessary to persist them.

Can i create "xnview.ini" in `User\AppData\Roaming\XnViewMP` directly?

**Comment by linsui** [comment]
If you just `"persist": "xnview.ini"`, xnviewmp will remove the hardlink and create a new file. `xnview.ini` in `$persist_dir` will not be changed and after update you wiil lose your configuration.

**Comment by Befod** [comment]
So if i create `xnview.ini` in `User\AppData\Roaming\XnViewMP` directly,is it conforms to the specification of scoop?

**Comment by linsui** [comment]
It's ok if you prefer nonportable app. Scoop prefer portable app so I'd like to make it portable if possible.:smile:

**Comment by Befod** [comment]
Thanks a lot.

File: bucket/xnviewmp.json [hunk 10]
```diff
@@ -0,0 +1,39 @@
+{
+    "homepage": "https://www.xnview.com/en/xnviewmp/",
+    "description": "Photo viewer, image resizer & batch converter.",
+    "version": "0.93.1",
+    "license": "Freeware",
```
Comment (rasa) [inline bucket/xnviewmp.json]:
```suggestion
    "license": "Freeware|Proprietary",
```
as license.txt says:
```
2. License

XnViewMP is provided as freeware for private (non commercial), or educational use, including non-profit organization (i.e. schools, universities, public authorities, police, fire brigade, and hospitals). 

In this case, you are granted the right to use and to make an unlimited number of copies of this software.

Company must purchase licenses to be able to use XnViewMP.
```

**Comment by Jogai** [comment]
Please merge 😄

**Pull request merged**